### PR TITLE
executable building system

### PR DESCRIPTION
**Pull request by ackagel**
Refactors project for building executables

Only tested on MacOS so far

### CODE REVIEW

**Comment by ackagel** [comment]
just want to make sure there isn't anything out of the ordinary here

File: AMP.spec [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# -*- mode: python ; coding: utf-8 -*-
+
+block_cipher = None
+
+
+a = Analysis(['src/main.py'],
+             pathex=['/Users/ackagel/Documents/dev/amp/AMP'],
+             binaries=[],
+             datas=[('/Users/ackagel/Documents/dev/amp/AMP/ui', 'ui')],
```
Comment (ngreenwald) [inline AMP.spec]: Is this going to cause problems?

**Review by ngreenwald — APPROVED** [review]
Looks good, just found one suspicious looking path

File: AMP.spec (hunk 1 above)
Comment (ackagel) [reply AMP.spec]: only if someone tries to build amp locally.  Either way, I'll make the path dynamically detected.

**Review by alex-l-kong — APPROVED** [review]
Looks good to me!

**Pull request merged**